x-pack/plugins/security_solution/public/detections/components/rules/rule_execution_status/utils.ts

```diff
@@ -5,35 +5,44 @@
  * 2.0.
  */
 
+import { IconColor } from '@elastic/eui';
 import { capitalize } from 'lodash';
+import { assertUnreachable } from '../../../../../common/utility_types';
 import { RuleExecutionStatus } from '../../../../../common/detection_engine/schemas/common';
 
-export const normalizeRuleExecutionStatus = (
-  value: RuleExecutionStatus | null | undefined
-): RuleExecutionStatus | null =>
-  value === RuleExecutionStatus['partial failure']
-    ? RuleExecutionStatus.warning
-    : value != null
-    ? value
-    : null;
+export const getStatusText = (value: RuleExecutionStatus | null | undefined): string | null => {
+  if (value == null) {
+    return null;
+  }
+  if (value === RuleExecutionStatus['partial failure']) {
+    return 'warning';
+  }
+  return value;
+};
```

Comment:
Clearer name, and easier to immediately grok without the nested ternary (same with `getStatusColor`), thanks you! 🎉